design-documents/graph-ql/coverage/Wishlist.graphqls

```diff
@@ -7,8 +7,9 @@ type Mutation {
 }
 
 type Customer {
-    wishlist: Wishlist! @doc(description: "Customer wishlist") # Commerce will extend filed with required `id` argument `wishlists(ids: ID!)`
-    wishlists: [Wishlist!]! @doc(description: "Customer multiple wishlists") # Multiple wishlists Commerce functionality
+    wishlist: Wishlist! @deprecated(reason: "Use `Customer.wishlists` or `Customer.wishlistByID") 
+    wishlistByID(id: ID!): Wishlist
```

Comment:
How about `v2` instead of `byId`, this would be more aligned with our current strategy.
```suggestion
    wishlist_v2(id: ID!): Wishlist
```

Reply:
I'm not sure prepending things with `_v2` by default is the best design we can do.

A field name should really describe to the client what the purpose of the field is.

- `wishlistByID(id: ID!)` - Makes it clear you're getting a wishlist by ID, but a bit redundant
- `wishlist(id: ID)` - Makes it clear you're getting a wishlist by ID
- `wishlist_v2(id: ID!)` - Makes it apparent that you're looking up by ID, but also makes you wonder why the v1 version of the field had issues.

I can see `_v2` in a scenario where we can't come up with a more descriptive name, but I think we have a variety of naming choices here. Can also be redundant and do `customer_wishlist` or `wishlist_details`. Plenty of options!

@nrkapoor mentioned recently that we have rules around how long a deprecation lasts before we're safe to remove a field. So the `_v2` will likely live long after the original is gone.

Reply:
Spoke with @nishant, and he's in favor of `_v2`, and I'm in favor of consensus :)

I'll push up changes